lib/errors/createErrorType.js

```diff
@@ -0,0 +1,67 @@
+var util = require("util");
+
+/**
+ * Creates a new type of error.
+ *
+ * @param {string} name - The error type name
+ * @param {string} message - The error message
+ * @param {string[]|function} additionalKeys - Key names to set from arguments (or function to manually do so)
+ *
+ * @returns {function} Returns an Error class
+ *
+ * @private
+ */
+function createErrorType(name, message, additionalKeys) {
+
+    // Create the error type
+    var NewErrorType = function() {
```

Comment:
I can see where you're going with the Error class centralization, though there's a (small) drawback to this approach. Even with setting the name as you are, you'll end up with NewErrorType as the error name in the console. The stack will have the right name, but I'm afraid it will negatively impact at-a-glance debugging.

Very sorry to do this to you, but would you mind re-decentralizing the classes?

Reply:
No problem. I thought I had made it safely through renaming the error type but you're right that it shows incorrectly in the console. Bummer. Are you confident there's no way around this or is it worth a little more research on my end?

Reply:
Nothing is turning up when I go hunting for ways to change that, unfortunately.